aea/skills/base.py

```diff
@@ -912,3 +765,348 @@ def _print_warning_message_for_non_declared_skill_components(
+            for component_id, component_config in by_id.items():
+                path = component_config.file_path
+                class_name = component_config.class_name
+                if path is not None:
+                    classes_in_path = component_classes_by_path[path]
+                    component_class_or_none: Optional[Type[SkillComponent]] = next(
+                        (
+                            actual_class
+                            for actual_class_name, actual_class in classes_in_path
+                            if actual_class_name == class_name
+                        ),
+                        None,
+                    )
+                    enforce(
+                        component_class_or_none is not None,
+                        self._get_error_message_prefix()
+                        + f"Cannot find class '{class_name}' for component '{component_id}' of type '{self._type_to_str(component_type)}' of skill '{self.configuration.public_id}' in module {path}",
+                    )
+                    component_class = cast(
+                        Type[SkillComponent], component_class_or_none
+                    )
+                    used_classes.add(component_class)
+                    result.append(
+                        _SkillComponentLoadingItem(
+                            component_id,
+                            component_config,
+                            component_class,
+                            component_type,
+                        )
+                    )
+                else:
+                    # process the configuration at the end of the loop
+                    not_resolved_configurations[
+                        (component_type, component_id)
+                    ] = component_config
+
+        for (component_type, component_id), component_config in copy(
+            not_resolved_configurations
+        ).items():
+            class_name = component_config.class_name
+            classes_by_type = class_index.get(class_name, {})
+            enforce(
+                class_name in class_index and component_type in classes_by_type,
+                self._get_error_message_prefix()
+                + f"Cannot find class '{class_name}' for skill component '{component_id}' of type '{self._type_to_str(component_type)}'",
+            )
+            classes = classes_by_type[component_type]
+            not_used_classes = classes.difference(used_classes)
+            enforce(
+                not_used_classes != 0,
+                f"Cannot find class of skill '{self.configuration.public_id}' for component configuration '{component_id}' of type '{self._type_to_str(component_type)}'.",
+            )
+            enforce(
+                len(not_used_classes) == 1,
+                self._get_error_message_ambiguous_classes(
+                    class_name, not_used_classes, component_type, component_id
+                ),
+            )
```

Comment:
given a configuration:
1. we first check if there are classes with the class name specified (1069)
2. then we check if the class is of the right type (1069)

also, this time we need to:
- filter the classes that have been used by the configurations which specified the `file_path` field (1076)
- check that there is only one way to select such class, because we cannot resolve the ambiguity by path this time (1080)